lib/config.js

```diff
@@ -220,13 +238,8 @@ function getLocalConfig(thisConfig, directory) {
         debug("Loading " + localConfigFile);
         localConfig = loadConfig(localConfigFile);
 
-        if (path.basename(localConfigFile) !== LOCAL_CONFIG_FILENAME) {
-
-            // Don't consider a local config file found if the package.json doesn't have the eslintConfig field.
-            if (!localConfig.hasOwnProperty(PACKAGE_CONFIG_FIELD_NAME)) {
-                continue;
-            }
-            localConfig = localConfig[PACKAGE_CONFIG_FIELD_NAME] || {};
+        if (!Object.keys(localConfig).length) {
```

Comment:
Oh wait, this is the replacement for lines 226-227.  Can you add a comment? Otherwise this is really unclear.